### DISCUSSION TITLE
Request: Option to disable nested matches

### DISCUSSION THREAD

**Discussion by AndydeCleyre** [espanso/espanso#1733] [Feature requests and Ideas]
For me, nested matches are only something that happen unintentionally, and I think are the reason I can't have a match/replacement like `:em`/`emulate`, at least while fast inject is disabled: https://github.com/espanso/espanso/issues/1701 

Aside from them triggering when I don't want them to, I don't understand the reasoning behind having a modified version of the trigger activate as if it were the full trigger ("nested" triggers the ":nested" trigger?). This is extra extra surprising. 

But even without that, I never want a nested match to be triggered. Some of my match definitions include modifying and pasting clipboard content, which can easily inadvertently include defined triggers.

So I'd love to either globally disable nested matches, or to be able to specify for each match definition that it should not trigger a secondary match (or like in the linked issue, an infinite number of matches). I would then add that to every match definition.

**Comment by smeech** [comment]
Sorry - I replied to your Issue: https://github.com/espanso/espanso/issues/1701#issuecomment-1826927527

I don't think what you're seeing is intentional.

I've tried all your examples (Linux Mint 21.2 x86_6, Xfce 4.18) and can't reproduce your outputs. I can only produce the nested match effect with the configuration specified in the [docs](https://espanso.org/docs/matches/basics/#nested-matches), where the nested match is triggered by:

    vars:
        params:
          trigger: :something
as shown in the docs example, so I don't get any accidental triggering.

I think what you're seeing is a bug.
